modules/sdk/gradle-plugins-change-log-builder/README.markdown

```diff
@@ -0,0 +1,90 @@
+# Change Log Builder Gradle Plugin
+
+The Change Log Builder Gradle plugin allows you to generate and maintain a
+change log file based on the Git commits on your project. A change log file
+generated by this plugin looks like this:
+
+```ini
+#
+# Bundle Version 1.0.1
+#
+9c77ff4c95cb1a325db3bdd089be105206e8b63c^..b421f00ac84b065685b131833fecc594fc01c760=LPS-123 LPS-1321
+
+#
+# Bundle Version 1.0.2
+#
+b421f00ac84b065685b131833fecc594fc01c760^..bc15d8d84e12b9544f78e4e3743c510dbaec2d89=LPS-456
+```
+
+Every time the [`buildChangeLog`](#buildchangelog) task is executed, a new line
+is added to the change log, which lists all Git [commit prefixes](#ticketidprefixes)
+(usually issue ticket IDs) that occurred in a certain range. The end of the
+range is always the tip of the current branch, while the start is:
+
+- the first time `buildChangeLog` is executed: the most recent commit from at least two years ago;
+- otherwise: the range end of the last line in the change log.
```

Comment:
@Ithildir I'm a little confused with the range description. I understand that the end range is always the tip of the current branch. The start range is a little hazy for me...

**First lets start with the first bullet:**
The first time `buildChangeLog` is executed, which must be a commit from over two years ago. (Is that the correct interpretation?)

**Second bullet:**

If the first commit using `buildChangeLog` is not a commit that's at least two years old, then use the last line in the change log. Why would the last line in the change log be the beginning of the range?

I think I'm misunderstanding. Can you describe this to me in different words? Maybe the light bulb will come on for me. Thanks!